client/helpers/scenes.ts

```diff
@@ -54,82 +103,188 @@ export interface PollSceneReadyArgs extends BaseArgs {
  */
-export async function createSceneWithSceneItems({
+export async function createSceneAndSceneItems({
   client,
   createSceneItemReqs,
   createSceneReq,
+  failFast,
   onMsg = console.log,
   onProgress,
   parallelism,
   polling,
   verbose,
-}: CreateSceneWithSceneItemsArgs): Promise<SceneData> {
-  const sceneId = (
-    await client.scenes.createScene({
-      createSceneRequest: createSceneReq(),
+}: CreateSceneAndSceneItemsReq): Promise<CreateSceneAndSceneItemsRes> {
+  const scene = (
+    await client.scenes.createScene({ createSceneRequest: createSceneReq() })
+  ).data;
+  const sceneId = scene.data.id;
+  const { a: queuedItems, b: errors } = partition(
+    (
+      await createSceneItems({
+        client,
+        createSceneItemReqs,
+        failFast: failFast ?? false,
+        onProgress,
+        parallelism,
+        sceneId,
+      })
+    ).queuedSceneItems,
+    (i: QueuedSceneItem) => !nullOrUndefined(i.job)
+  );
+
+  if (queuedItems.length === 0) return { errors, scene };
+
+  await pollQueuedJob<SceneItem>({
+    id: (queuedItems[queuedItems.length - 1].job as QueuedJob).data.id,
+    getQueuedJob: (id) => client.sceneItems.getQueuedSceneItem({ id }),
+    allow404: true,
+    polling,
+  });
+
+  if (verbose) onMsg(`Committing scene and polling until ready...`);
+
+  await updateScene({
+    attributes: { state: UpdateSceneRequestDataAttributesStateEnum.Commit },
+    client,
+    sceneId,
+  });
+  await pollSceneReady({ client, id: sceneId, onMsg, polling, verbose });
+
+  if (verbose) onMsg(`Fitting scene's camera to scene-items...`);
+
+  const updated = (
+    await updateScene({
+      attributes: { camera: { type: CameraFitTypeEnum.FitVisibleSceneItems } },
+      client,
+      sceneId,
     })
-  ).data.data.id;
+  ).scene;
+  return { errors, scene: updated };
+}
+
+/**
+ * Create scene items within a scene.
+ */
+export async function createSceneItems({
+  client,
+  createSceneItemReqs,
+  failFast,
+  onProgress,
+  parallelism,
+  sceneId,
+}: CreateSceneItemsReq): Promise<CreateSceneItemsRes> {
   const limit = pLimit(parallelism);
   let complete = 0;
-  const responses = await Promise.all(
+  const queuedSceneItems = await Promise.all(
     createSceneItemReqs.map((req) =>
-      limit<CreateSceneItemRequest[], AxiosResponse<QueuedJob>>(
+      limit<CreateSceneItemRequest[], QueuedSceneItem>(
         async (r: CreateSceneItemRequest) => {
-          const res = await client.sceneItems.createSceneItem({
-            id: sceneId,
-            createSceneItemRequest: r,
-          });
+          let job: QueuedJob | undefined;
+          let failure: Failure | undefined;
+          try {
+            job = (
+              await client.sceneItems.createSceneItem({
+                id: sceneId,
+                createSceneItemRequest: r,
+              })
+            ).data;
+          } catch (error) {
+            if (failFast) throw error;
```

Comment:
Throw on `failFast` (the default CLI behavior), but allow clients to pass `false` to instead return a list of errors.